src/workflow/ArcanistLandWorkflow.php

```diff
@@ -256,7 +272,13 @@ public function run() {
 
     $engine = null;
     if ($this->isGit && !$this->isGitSvn) {
-      $engine = new ArcanistGitLandEngine();
+      if ($this->shouldUseSubmitQueue && !$this->tbr) {
```

Comment:
Avoid change existing code. Just add your your lines below:

```
$engine = new ArcanistGitLandEngine();
if ($this->shouldUseSubmitQueue && !$this->tbr) {
  $engine = new ArcanistSubmitQueueEngine(
  $this->submitQueueClient,
  $this->getConduit());
}
```

This is not "as effective" but might be easier merge upstream changes later.